uber/models/attendee.py

```diff
@@ -826,33 +833,74 @@ def hour_map(self):
                 all_hours[hour] = shift.job
         return all_hours
 
+    @property
+    def job_visibility(self):
+        from uber.models.department import Job
+        return Job.ALL_STAFF if self.badge_type == c.STAFF_BADGE \
+            else Job.ALL_VOLUNTEERS
+
+    @cached_property
+    def available_job_filters(self):
```

Comment:
The `available_jobs()` and `possible()` methods now run several orders of magnitude faster because we've removed several large joins.

Reply:
I fundamentally changed the structure of the query. Instead of doing a table join in the database, I am checking if the id exists in a given array of ids, using the `IN` operator. Thus, we're only looking at a single indexed column of a single table.

Reply:
I should add that it's okay to use an `IN` operator in this case, because the number of departments in our system will always be constrained by the limits of reality. We probably won't have hundreds of thousands of departments, and even if we did, no one will ever be a member of hundreds of departments.

You _could_ pass an array of thousands of ids to the `IN` operator, and I believe it would still work. At that point, I believe postgres creates a temp table for the ids and ends up doing a join anyway, so the worst case is about the same as the original query.